components/nexusoperations/workflow/commands.go

```diff
@@ -134,6 +134,7 @@ func (ch *commandHandler) HandleScheduleCommand(
 				WorkflowTaskCompletedEventId: workflowTaskCompletedEventID,
 			},
 		}
+		he.UserMetadata = command.UserMetadata
```

Comment:
@bergundy instead of copying the UserMetadata in every command handler here, I was thinking about exposing the event from the command handlers and copying the metadata in the caller function (`handleCommand()`). This way if we add more commands later, we don't have to remember to copy metadata in each of them.
But that needs a bigger refactor of the command handler since the function signature changes. Thoughts on doing that despite the work involved (we need to change a lot of tests as well)?

Reply:
Generally sounds good to me, you could do a simple interface check on the command.
To actually set `UserMetadata` you'll likely need to use reflection.

Reply:
Oh, you actually don't need this, I thought the metadata is on the attributes not on the top level message.